interface/src/error.rs

```diff
@@ -1,70 +1,168 @@
+#[cfg(feature = "codama")]
+use codama_macros::CodamaErrors;
 use {
     num_traits::{FromPrimitive, ToPrimitive},
     solana_program_error::ProgramError,
 };
 
 /// Reasons the Stake might have had an error.
+#[cfg_attr(feature = "codama", derive(CodamaErrors))]
+#[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(test, derive(strum_macros::FromRepr, strum_macros::EnumIter))]
 #[cfg_attr(
     feature = "serde",
     derive(serde_derive::Deserialize, serde_derive::Serialize)
 )]
-#[derive(Clone, Debug, PartialEq, Eq)]
 pub enum StakeError {
     // 0
     /// Not enough credits to redeem.
+    #[cfg_attr(
+        feature = "codama",
+        codama(error(code = 0, message = "Not enough credits to redeem"))
```

Comment:
I think the error codes are automatically derived from enum position. Aka, can we remove the `code = #`?

Reply:
They are; we can yank them. Done in 911b925